src/pytest_mypy/__init__.py

```diff
@@ -234,6 +240,14 @@ def runtest(self) -> None:
                 error.partition(":")[2].partition(":")[0].strip() == "note"
                 for error in errors
             ):
+                if self.session.config.option.mypy_xfail:
+                    self.add_marker(
+                        pytest.mark.xfail(
+                            raises=MypyError,
+                            reason="The mypy errors in this file"
+                            " were xfailed by --mypy-xfail.",
```

Comment:
Nit: The wording feels a little verbose, especially considering pytest also emits "XFAIL" in the output. Maybe rephrase for conciseness?

```suggestion
                            reason="Expected mypy errors in this file.",
```
or
```suggestion
                            reason="Expected mypy errors in this file (--mypy-xfail).",
```

(similar for status)